dsf-bpe/dsf-bpe-server/src/main/java/org/highmed/dsf/bpe/spring/config/MasterPatientIndexConfig.java

```diff
@@ -0,0 +1,27 @@
+package org.highmed.dsf.bpe.spring.config;
+
+import org.highmed.mpi.client.MasterPatientIndexClientFactory;
+import org.highmed.mpi.client.MasterPatientIndexClientServiceLoader;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class MasterPatientIndexConfig
+{
+	@Value("${org.highmed.dsf.bpe.mpi.webservice.factory.class}")
+	private String masterPatientIndexClientFactoryClass;
+
+	@Bean
+	public MasterPatientIndexClientServiceLoader masterPatientIndexClientServiceLoader()
+	{
+		return new MasterPatientIndexClientServiceLoader();
+	}
+
+	@Bean
+	public MasterPatientIndexClientFactory masterPatientIndexClientFactory()
+	{
+		return masterPatientIndexClientServiceLoader()
+				.getMasterPatientIndexClientFactory(masterPatientIndexClientFactoryClass);
```

Comment:
```suggestion
		return masterPatientIndexClientServiceLoader()
				.getMasterPatientIndexClientFactory(masterPatientIndexClientFactoryClass).get()
				.orElseThrow(() -> new NoSuchElementException("Master patient index client factory with classname='" + masterPatientIndexClientFactoryClass+ "' not found"));
```

See MasterPatientIndexClientFactory comment. No logging needed since the NoSuchElementException will result in the SpringContext failing to be initialized.